benchmarks/nvector/kokkos/test_nvector_performance_kokkos.cpp

```diff
@@ -0,0 +1,267 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the testing routine to check the performance of the
+ * NVECTOR Kokkos module implementation.
+ * -----------------------------------------------------------------*/
+
+#include <nvector/nvector_kokkos.hpp>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sundials/sundials_math.h>
+
+#include "test_nvector_performance.h"
+
+#if defined(USE_CUDA)
+using ExecSpace = Kokkos::DefaultExecutionSpace;
+// using ExecSpace = Kokkos::Cuda;
+#elif defined(USE_HIP)
+#if KOKKOS_VERSION / 10000 > 3
+using ExecSpace = Kokkos::HIP;
+#else
+using ExecSpace = Kokkos::Experimental::HIP;
+#endif
+#elif defined(USE_OPENMP)
+using ExecSpace = Kokkos::OpenMP;
+#else
+using ExecSpace = Kokkos::Serial;
+#endif
+
+using VecType  = sundials::kokkos::Vector<ExecSpace>;
+using SizeType = VecType::size_type;
+
+/* private functions */
+static int InitializeClearCache(sunindextype cachesize);
+static int FinalizeClearCache();
+
+/* private data for clearing cache */
+struct Cache
+{
+  Kokkos::View<sunrealtype*, ExecSpace::memory_space> device;
+  Kokkos::View<sunrealtype*, ExecSpace::memory_space>::HostMirror host;
+};
+
+static Cache* cache;
+
+/* ----------------------------------------------------------------------
+ * Main NVector Testing Routine
+ * --------------------------------------------------------------------*/
+int main(int argc, char* argv[])
+{
+  sundials::Context sunctx;
+  int flag;            /* return flag     */
+
+  /* CLI args */
+  sunindextype veclen;     /* vector length   */
```

Comment:
@gardner48 Should veclen match SizeType since that's what the constructor asks for? I'm not sure what the advantage is to use sunindextype for it and just cast it to SizeType later.

Reply:
This could be `SizeType` but we'd need to swap some occurrences of `veclen` below to `static_cast<sunindextype>(veclen)` to avoid the conversion warning e.g., `flag = Test_N_VLinearSum(X, static_cast<sunindextype>(veclen), ntests);`